src/Data/Vector/Generic/Sized.hs

```diff
@@ -1928,3 +1926,10 @@ instance (VG.Vector v a, Floating a, KnownNat n) => Floating (Vector v n a) wher
     asinh   = map asinh
     acosh   = map acosh
     atanh   = map atanh
+
+instance (VG.Vector v a, Binary a, KnownNat n) => Binary (Vector v n a) where
```

Comment:
I would write this instance differently to parallel the instances for `(a,a)`, `(a,a,a)`, etc.

```
instance Binary (Vector v n a) where
    get = replicateM Data.Binary.get
    put = mapM_ put
```

There is no reason to store or check the length, since the length is already in the type of the vector you are trying to deserialize.

Reply:
Makes sense - I suppose I was following the ``[a]`` instance a bit too literally.